lib/mozilla-ast.js

```diff
@@ -1158,11 +1163,18 @@
 
     var FROM_MOZ_STACK = null;
 
-    function from_moz(node) {
-        FROM_MOZ_STACK.push(node);
-        var ret = node != null ? MOZ_TO_ME[node.type](node) : null;
+    function from_moz(moz) {
+        FROM_MOZ_STACK.push(moz);
+        var node = null;
+        if (moz) {
+            if (!HOP(MOZ_TO_ME, moz.type)) {
+                var s = my_start_token(moz);
+                js_error("Unsupported type: " + moz.type, s.file, s.line, s.col, s.pos);
+            }
+            node = MOZ_TO_ME[moz.type](moz);
+        }
         FROM_MOZ_STACK.pop();
-        return ret;
+        return node;
```

Comment:
@kzc this is https://github.com/mishoo/UglifyJS/pull/4870#issuecomment-827012910